src/horust/bus.rs

```diff
@@ -6,8 +6,41 @@
 //!
 
 use crossbeam::channel::{unbounded, Receiver, Sender};
-use std::fmt::Debug;
 use std::fmt::Formatter;
+use std::{
+    fmt::Debug,
+    sync::{Arc, Mutex},
+};
+
+#[derive(Clone)]
+pub struct SharedState<T>
+where
+    T: Clone,
+{
+    /// Bus input - sender side
+    sender: Sender<Message<T>>,
+    /// Bus output - all the senders
+    senders: Arc<Mutex<Vec<(u64, Sender<Message<T>>)>>>,
+}
+
+impl<T> SharedState<T>
+where
+    T: Clone,
+{
+    /// Add another connection to the bus
+    pub fn join_bus(&self) -> BusConnector<T> {
+        let mut senders = self.senders.lock().unwrap();
+
+        let (sender, receiver) = unbounded();
+
+        let id = senders.len() as u64;
+        senders.push((id, sender));
+
+        drop(senders);
```

Comment:
I'm curious, why do you need the explicit drop? to release the lock before the self.clone?

Reply:
>  to release the lock before the self.clone?

Well, yes.  That was my intuition - however, I checked on rust playground, and it's not necessary.

Reply:
Cool, I knew that locks are released as soon as they go out of scope, but I wasn't 100% sure with the self.clone and an Arc, I thought that maybe weird things could happen without explicit dropping before cloning...